app/services/partner/submission_input_validator.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+module Partner
+  # Validate submission input (XML) from project partners
+  class SubmissionInputValidator
+    PHD_REGEX = /p\W*h\W*d/i
+    ENG_REGEX = /^eng$/i
+
+    def initialize(**params)
+      @params = params
+    end
+
+    def parse!
+      @result = Contract.new.call(params)
+    end
+
+    def valid?
+      result.errors.blank?
+    end
+
+    def error_message
+      result.errors(full: true).messages.join('; ')
+    end
+
+    def parsed # rubocop:disable Metrics/AbcSize
+      @parsed ||= result.to_h.fetch(:DISSERTATION).tap do |hash|
+        hash[:dissertation_id] = hash.delete(:dissertationid)
+        hash[:etd_type] = hash.delete(:type)
+        hash[:provost] = hash.delete(:vpname)
+        hash[:ps_career] = hash.delete(:career)
+        hash[:department] = hash.delete(:program)
+        hash[:major] = hash.delete(:plan)
+        hash[:ps_subplan] = hash.delete(:subplan) if hash.key?(:subplan)
+        hash[:readers] = hash.delete(:reader)
+        if (deadline = hash.dig(:sub, :deadline)).present?
+          hash[:sub] = deadline
+        end
```

Comment:
Here's where we massage the keys we get from the registrar and map them into this application's style.